File: Proyecto con oracle/ClienteComputador/ClienteComputador/ClienteComputador/Periferico.cs
namespace ClienteComputador
{
    public class Periferico
    {
        public int id { get; set; }
        public string nombre { get; set; } = "";
        public string tipo { get; set; } = "";
        public bool activo { get; set; }

        public string ActivoStr => activo ? "Activo" : "Inactivo";
    }
}

File: Proyecto con oracle/cliente-react-emp/src/Periferico.js
export class Periferico {

  constructor(id, nombre, tipo, activo) {
    this.id = id;
    this.nombre = nombre;
    this.tipo = tipo;
    this.activo = activo;
  }

  toJson() {
    var datos = {
      id: this.id,
      nombre: this.nombre,
      tipo: this.tipo,
      activo: this.activo
    };
    return datos;
  }

  static fromJson(data) {
    var periferico = new Periferico(
      data.id,
      data.nombre,
      data.tipo,
      data.activo
    );
    return periferico;
  }
}
